### DISCUSSION TITLE
[TS0601_thermostat] "No converter available for"

### DISCUSSION THREAD

**Discussion by mcarni** [Koenkk/zigbee2mqtt#14436] [Devices]
probably a noob question, i wrote on the discord Off-Topic but I guess it was the wrong place

I just bought a thermostat valve [Tuya TS0601)](https://www.zigbee2mqtt.io/devices/TS0601_thermostat.html) and i am starting to play with zigbee2mqtt and mosquitto
i can control the valve, for example i can set the current_heating_setpoint to 11°C with: 
`mosquitto_pub -t 'zigbee2mqtt/BG Heater/set' -m '{"running_state": "11"}'' 

but i cannot read the value, if i do 
'mosquitto_pub -t 'zigbee2mqtt/BG Heater/get' -m '{"running_state": ""}'' 
i get nothing, just an "error 2022-10-11 08:54:27: No converter available for 'get' 'current_heating_setpoint' ()" 

i am guessing that I am doing something noobish, any help would be appreciated.

**Comment by Koenkk** [comment]
Reading values is not supported for TuYa devices (their API doesn't have such feature)

**Comment by mcarni** [comment]
Thanks

I thought reading some values was possible, probably i got confused by the wiki for this device:
"To read send a message to zigbee2mqtt/FRIENDLY_NAME/get with payload {"current_heating_setpoint": ""}."

One more suggestion if possible...
I was playing with mosquitto and I see that I can mosquitto_sub to the device and get a json with all the property values. ..
Since the device is publishing its state every time a value is set, I guess I could try to make a nodejs app that calls for mosquitto_sub to the device and get the values from there. 
Would this be a dumb workaround?  Are there better ways?
I don't need all properties...maybe 1 or 2

Thanks a lot

M

**Reply by Koenkk** [reply]
If you always want to retrieve the latest device state upon connection you can use the MQTT retain feature:

- https://www.hivemq.com/blog/mqtt-essentials-part-8-retained-messages/
- https://www.zigbee2mqtt.io/guide/configuration/devices-groups.html#common-device-options

**Comment by mcarni** [comment]
thanks a lot... i will look into the retain feature

thanks M
